src/main/kotlin/platform/sponge/creator/SpongeProjectConfig.kt

```diff
@@ -47,14 +49,31 @@ class SpongeProjectConfig : ProjectConfig(), MavenCreator, GradleCreator {
         module: Module,
         buildSystem: MavenBuildSystem
     ): ProjectCreator {
-        return SpongeMavenCreator(rootDirectory, module, buildSystem, this)
+        val apiVersion = SemanticVersion.parse(spongeApiVersion)
+        return if (apiVersion < SpongeConstants.API8) {
+            SpongeMavenCreator(rootDirectory, module, buildSystem, this)
+        } else {
+            Sponge8MavenCreator(rootDirectory, module, buildSystem, this)
+        }
     }
 
     override fun buildGradleCreator(
         rootDirectory: Path,
         module: Module,
         buildSystem: GradleBuildSystem
     ): ProjectCreator {
-        return SpongeGradleCreator(rootDirectory, module, buildSystem, this)
+        val apiVersion = SemanticVersion.parse(spongeApiVersion)
+        return if (apiVersion < SpongeConstants.API8) {
+            SpongeGradleCreator(rootDirectory, module, buildSystem, this)
+        } else {
+            Sponge8GradleCreator(rootDirectory, module, buildSystem, this)
+        }
+    }
+
+    override fun configureRootGradle(rootDirectory: Path, buildSystem: GradleBuildSystem) {
+        val apiVersion = SemanticVersion.parse(spongeApiVersion)
+        if (apiVersion >= SpongeConstants.API8) {
+            buildSystem.gradleVersion = SemanticVersion.release(7, 0, 2)
```

Comment:
what is the default Gradle version? I believe SpongeGradle should work fine with gradle 6.x if projects prefer that

Reply:
The creator uses 5.6.1 by default which is quite old. (I plan to update most templates and the default Gradle version later)

That and I'd rather stick with what the official template uses.